src/main/host/process.rs

```diff
@@ -540,14 +539,20 @@ impl Process {
             return;
         }
 
-        let threads = self.threads.borrow();
-        let Some(thread) = threads.get(&tid) else {
-            debug!("Thread {} no longer exists", tid);
-            return;
+        // Clone the thread reference, so that we don't hold a reference to the thread
+        // list while running the thread.
+        let threadrc = {
+            let threads = self.threads.borrow();
+            let Some(thread) = threads.get(&tid) else {
+                debug!("Thread {} no longer exists", tid);
+                return;
+            };
+            thread.clone(host.root())
```

Comment:
Moved the explanatory comment closer and tweaked the wording a bit - we clone the reference here so that we're not holding a dynamically borrowed reference to the thread list while the thread is running.